### PR TITLE
[homekit] change SECURED=OFF for HomeKit lock

### PR DESCRIPTION
**Pull request by ccutrer**
since this is what most real locks (especially ZWave ones) do

Signed-off-by: Cody Cutrer <cody@cutrer.us>

cc @yfre

### CODE REVIEW

**Comment by ccutrer** [comment]
PR updated to also support JAMMED state

**Comment by yfre** [comment]
@blafois you did the initial contribution. does mapping OFF/ON fit to your device as well? or do we need to make it configurable?

**Comment by blafois** [comment]
Just on-off works ! You might probably support as well an "inverted" parameter - depending on implementations (ON=locker or ON=unlocked). Mine is not a problem as I use an MQTT bridge, so I can configure at MQTT conversation.

**Comment by yfre** [comment]
ok. we keep PR as it is. we can add configurability later to that and other accessories at once, option "inverted" can be interesting for all switch and contact items. 

this PR looks good for me. but we need a second reviewer.
@cpmeister @J-N-K can you please take a look on this PR.

File: bundles/org.openhab.io.homekit/src/main/java/org/openhab/io/homekit/internal/accessories/HomekitLockImpl.java [hunk 1]
```diff
@@ -47,10 +50,14 @@ public HomekitLockImpl(HomekitTaggedItem taggedItem, List<HomekitTaggedItem> man
...
-        if (state != null) {
+
+        final State state = getItem(HomekitCharacteristicType.LOCK_CURRENT_STATE, GenericItem.class).getState();
+        if (state instanceof DecimalType) {
+            return CompletableFuture.completedFuture(LockCurrentStateEnum.fromCode(
+                state.as(DecimalType.class).intValue()));
+        } else if (state instanceof OnOffType) {
             return CompletableFuture.completedFuture(
-                    state == OnOffType.OFF ? LockCurrentStateEnum.SECURED : LockCurrentStateEnum.UNSECURED);
+                state.equals(OnOffType.ON) ? LockCurrentStateEnum.SECURED : LockCurrentStateEnum.UNSECURED);
```
Comment (J-N-K) [inline bundles/org.openhab.io.homekit/src/main/java/org/openhab/io/homekit/internal/accessories/HomekitLockImpl.java:60]:
```suggestion
                state == OnOffType.ON ? LockCurrentStateEnum.SECURED : LockCurrentStateEnum.UNSECURED);
```

**Comment by J-N-K** [comment]
DCO failed due to "commit suggestion", manually set to pass

**Comment by J-N-K** [comment]
Trais fails unrelated (could not resolve a part of the codestyle checks)

**Pull request merged**